File: lab2/source/point.cpp
#include "operations.h"
#include "factory.h"

class Point : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        if(stack.empty()){
            throw std::runtime_error("stack is empty");
        }
        std::cout<< stack.top() << std::endl;
        stack.pop();
    }
};

namespace {
    Operations* createPoint() {
        return new Point;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string("."), createPoint);
}

File: jaba/lab1/src/main/java/org/example/Point.java
package org.example;

import java.io.OutputStream;
import java.util.Scanner;
import java.util.Stack;

public class Point extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s) {
        emptyCheck(stack);
        System.out.println(stack.pop());
    }
}
